internal/jsonnum/jsonnum.go

```diff
@@ -2,7 +2,7 @@ package jsonnum
 
 // ReadNumber returns the index of the end of the number value
 // and err=true if a syntax error was encountered.
-func ReadNumber[S ~string | ~[]byte](s S) (trailing S, err bool) {
```

Comment:
This boolean actually indicates whether there was an error, please don't change it in `internal/jsonnum`